RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractPrefetch.java

```diff
@@ -0,0 +1,379 @@
+    private static final String PREFETCH_OUTPUT_FILE_NAME = "Output.txt"; //NON-NLS
+    private static final String PREFETCH_ERROR_FILE_NAME = "Error.txt"; //NON-NLS
+    private static final String PREFETCH_PARSER_DB_FILE = "Autopsy_PF_DB.db3"; //NON-NLS
+    private static final String PREFETCH_DIR_NAME = "prefetch"; //NON-NLS
+
+    @Messages({
+        "ExtractPrefetch_module_name=Windows Prefetch Extractor"
+    })
+    ExtractPrefetch() {
+        this.moduleName = Bundle.ExtractPrefetch_module_name();
+    }
+
+    @Override
+    void process(Content dataSource, IngestJobContext context, DataSourceIngestModuleProgress progressBar) {
+
+        this.context = context;
+
+        String modOutPath = Case.getCurrentCase().getModuleDirectory() + File.separator + PREFETCH_DIR_NAME;
+        File dir = new File(modOutPath);
+        if (dir.exists() == false) {
+            dir.mkdirs();
+        }
+        
+        extractPrefetchFiles(dataSource);
+        
+        final String prefetchDumper = getPathForPrefetchDumper();
+        if (prefetchDumper == null) {
+            logger.log(Level.SEVERE, "Error finding parse_prefetch program"); //NON-NLS
+            return; //If we cannot find the parse_prefetch program we cannot proceed
+        }
+
+        if (context.dataSourceIngestIsCancelled()) {
+            return;
+        }
+
+        String modOutFile = modOutPath + File.separator + PREFETCH_PARSER_DB_FILE;
+        try {
+            String tempDirPath = RAImageIngestModule.getRATempPath(Case.getCurrentCase(), PREFETCH_DIR_NAME );
+            parsePrefetchFiles(prefetchDumper, tempDirPath, modOutFile, modOutPath);
+            createAppExecArtifacts(modOutFile);
+        } catch (IOException ex) {
+            logger.log(Level.WARNING, "Error runing parse_prefetch or creating artifacts.", ex); //NON-NLS             
+        }
+    }
+    
+    /**
+     * Extract prefetch file to temp directory to process.  Checks to make sure that the prefetch files only
+     * come from the /Windows/Prefetch directory
+     * 
+     * @param dataSource - datasource to search for prefetch files
+     * 
+     */
+
+    void extractPrefetchFiles(Content dataSource) {
+        List<AbstractFile> pFiles;
+
+        FileManager fileManager = Case.getCurrentCase().getServices().getFileManager();
+
+        try {
+            pFiles = fileManager.findFiles(dataSource, "%.pf"); //NON-NLS            
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, "Unable to find prefetch files.", ex); //NON-NLS
+            return;  // No need to continue
+        }
+
+        for (AbstractFile pFile : pFiles) {
+
+            if (context.dataSourceIngestIsCancelled()) {
+                return;
+            }
+
+            String prefetchFile = RAImageIngestModule.getRATempPath(Case.getCurrentCase(), "prefetch") + File.separator + pFile.getName();
```

Comment:
I think this should use PREFETCH_DIR_NAME instead

Reply:
Good catch.  Fixed.